app/src/editor/shared/components/Client.jsx

```diff
@@ -0,0 +1,113 @@
+/**
+ * Provides a shared streamr-client instance.
+ */
+
+/* eslint-disable react/no-unused-state */
+
+import React, { Component } from 'react'
+import { connect } from 'react-redux'
+import t from 'prop-types'
+import StreamrClient from 'streamr-client'
+import { selectAuthApiKeyId } from '$shared/modules/resourceKey/selectors'
+import { getMyResourceKeys } from '$shared/modules/resourceKey/actions'
+
+import * as services from '../services'
+
+export const ClientContext = React.createContext()
+
+class ClientProviderComponent extends Component {
+    static propTypes = {
+        authKey: t.string,
+    }
+
+    componentDidMount() {
+        this.setup()
+    }
+
+    componentDidUpdate() {
+        this.setup()
+    }
+
+    componentWillUnmount() {
+        this.teardown()
+    }
+
+    setup() {
+        const { authKey } = this.props
+        if (!authKey || this.state.client) { return }
+        this.setState({
+            client: new StreamrClient({
+                url: process.env.STREAMR_WS_URL,
+                authKey,
```

Comment:
We are using `streams-client` v.0.12.2, this should be `apiKey` (https://github.com/streamr-dev/streamr-client-javascript/tree/v0.12.2). But if it works, probably best to leave it and update to client 1.0.0 at some point.

Reply:
Yeah it seems to work.